dowhy/timeseries/temporal_shift.py

```diff
@@ -23,43 +22,51 @@ def find_lagged_parent_nodes(graph:nx.DiGraph, node:str) -> Tuple[List[str], Lis
             time_lags.append(edge_data['time_lag'])
     return parent_nodes, time_lags
 
-# once we have the parent dictionary then we can parse it and shift columns within the dataframe with the appropriate lag
 def shift_columns(df: pd.DataFrame, columns: List[str], lag: List[int]) -> pd.DataFrame:
```

Comment:
maybe a more informative name is `create_lagged_columns` or `shift_columns_by_lag`? 
How do we handle the case when there are multiple lags specified for the same pair of nodes? can you add a test for that?